File: 1170. Compare Strings by Frequency of the Smallest Character/1170.py
class Solution:
    def numSmallerByFrequency(self, queries, words):
        for i in range(len(queries)):
            queries[i] = self.countFrequency(queries[i])

        for i in range(len(words)):
            words[i] = self.countFrequency(words[i])

        words.sort(reverse = True)

        relist = list()
        for i in queries:
            count = 0
            for j in words:
                if j > i:
                    count+=1
                else:
                    break
            relist.append(count)
        
        return relist

    def countFrequency(self, word: str) -> int:
        mylist = [0]*26
        for i in word:
            mylist[ord(i)-97] += 1

        for i in mylist:
            if i != 0:
                return i

    # 写的很简洁很干练啊...
    def solutionOne(self, queries, words):
        w_counts = sorted([w.count(min(w)) for w in words])
        q_counts = [q.count(min(q)) for q in queries]
        return [sum( q_count < w_count for w_count in w_counts) for q_count in q_counts]


a = Solution()
print(a.numSmallerByFrequency(["bbb","cc"], ["a","aa","aaa","aaaa"]))

File: 1170. Compare Strings by Frequency of the Smallest Character/1170.go
package leetcode

import "sort"

func numSmallerByFrequency(queries []string, words []string) []int {
	queriesNum := make([]int, len(queries))
	wordsNum := make([]int, len(words))
	for i:=0; i<len(queries); i++{
		queriesNum[i] = countFrequency(queries[i])
	}
	for i:=0; i<len(words); i++{
		wordsNum[i] = countFrequency(words[i])
	}
	
	sort.Sort(sort.Reverse(sort.IntSlice(wordsNum)))
	for i, vq := range queriesNum{
		var count int
		for _, vw := range wordsNum{
			if vw>vq{
				count ++
			}else{
				break
			}
		}
		queriesNum[i] = count
	}
	return queriesNum
}

func countFrequency(words string) int{
	myslice := make([]int, 26)
	for _,i := range words{
		myslice[i-'a']++
	}
	for _,i := range myslice{
		if i != 0{
			return i
		}
	}
	return 0
}

// 这个用了search的方法, 效果基本一样
func numSmallerByFrequency1(queries []string, words []string) []int {
    var ans []int
    count := make([]int, len(words))
    for i, w := range words {
        count[i] = countFrequency(w)
    }
    sort.Ints(count)
    for _, q := range queries {
        ans = append(ans, len(count) - sort.SearchInts(count, countFrequency(q)+1))
    }
    return ans
}